operator/pkg/client/factory_test.go

```diff
@@ -83,6 +95,115 @@ func (f *fakeObject) DeepCopyObject() runtime.Object {
 	return f
 }
 
+func TestIntegrationFactoryOperatorV1(t *testing.T) {
+	testutil.SkipIfNotIntegration(t)
+
+	scheme := controller.V1Scheme
+
+	env := testenv.New(t, testenv.Options{
+		Scheme: scheme,
+		CRDs:   crds.All(),
+		Logger: testr.New(t),
+	})
+
+	var clientFactory *Factory
+	var r *vectorized.ClusterReconciler
+
+	c := env.Client()
+	err := c.Create(context.Background(), &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: env.Namespace(),
+		},
+	})
+	require.NoError(t, err)
+
+	err = c.Create(context.Background(), &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: env.Namespace(),
+		},
+		Subjects: []rbacv1.Subject{
+			{Kind: "ServiceAccount", Namespace: env.Namespace(), Name: "test"},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "cluster-admin",
+		},
+	})
+	require.NoError(t, err)
+
+	env.SetupManager("test", func(mgr ctrl.Manager) error {
+		dialer := kube.NewPodDialer(mgr.GetConfig())
+		clientFactory = NewFactory(mgr.GetConfig(), mgr.GetClient()).WithDialer(dialer.DialContext)
+
+		r = &vectorized.ClusterReconciler{
+			Client:                mgr.GetClient(),
+			Log:                   testr.New(t),
+			AdminAPIClientFactory: admin.NewNodePoolInternalAdminAPI,
+			Dialer:                dialer.DialContext,
+			Scheme:                scheme,
+		}
+		r.WithConfiguratorSettings(resources.ConfiguratorSettings{
```

Comment:
We don't want this to be hard coded long term buuuuut we don't currently have infra for relying on containers build from the local repo in integration tests... Mind adding a TODO here so this doesn't get copy pasted too much?

Reply:
done